momepy/elements.py

```diff
@@ -912,46 +912,56 @@ def get_network_ratio(df, edges, initial_buffer=500):
 
     """
 
-    # intersection-based join
-    buff = edges.buffer(0.01)  # to avoid floating point error
-    inp, res = buff.sindex.query_bulk(df.geometry, predicate="intersects")
+    (df_ix, edg_ix), dist = edges.sindex.nearest(
+        df.geometry, max_distance=initial_buffer, return_distance=True
+    )
+
+    touching = dist < 0.1
+
     intersections = (
-        df.iloc[inp]
-        .reset_index(drop=True)
-        .intersection(buff.iloc[res].reset_index(drop=True))
+        df.iloc[df_ix[touching]]
+        .intersection(edges.buffer(0.0001).iloc[edg_ix[touching]], align=False)
```

Comment:
No, this is only a correction of floating point issue. Ideally, we would do the intersection with the non-buffered linestring and use length of the intersection instead of area but in reality, this leads to a lot of empty linestrings and similar issues. That is why I'm doing this. So no need to expose this.